app/src/main/java/com/andrew/tasky/presentation/agenda_item_detail/AgendaItemDetailFragment.kt

```diff
@@ -89,32 +92,56 @@ class AgendaItemDetailFragment: Fragment(R.layout.fragment_agenda_item_detail) {
         }
     }
 
+    private fun setupAgendaInitialViewModels() {
+        if (!viewModel.isInitiallySetup.value) {
+            args.agendaItem?.let {
+                viewModel.setTitle(it.title)
+                viewModel.setDescription(it.description)
+                viewModel.setSelectedDate(it.startDateAndTime.toLocalDate())
+                viewModel.setSelectedTime(it.startDateAndTime.toLocalTime())
+                viewModel.setIsDone(it.isDone)
+                viewModel.setSelectedReminderTime(it.reminderTime)
+            }
+            viewModel.setEditMode(args.isInEditMode)
+            viewModel.setInitialSetupToTrue()
+        }
+    }
+
     private fun setupAgendaItemType(){
```

Comment:
In general, I'd avoid this approach of having one screen for multiple types of agenda items. It's true that you can avoid a bit of duplicate code like this, but that shouldn't always be the highest priority.

Imagine at some point, requirements change and just the event detail screen will look a lot different than task & reminder screen. That would mean, you'd need to restructure this whole big fragment and create a separate screen for events. This is just not maintainable in the long run.

Instead, I'd create reusable UI components (in separate XML files, for example for the reminder section of an agenda item) which you can then use in all of your separate screens (EventFragment, TaskFragment, ReminderFragment). 
If you're unsure about how to create such reusable UI components, let me know :)